src/components/product/index.jsx

```diff
@@ -0,0 +1,198 @@
+import React, { useEffect, useState } from "react";
+import { useParams, useNavigate } from "react-router-dom";
+import { getProductComments, getProductDetails } from "api";
+import styles from "./product.module.css";
+import Header from "components/Header";
+
+function ProductDetail() {
+  const { productId } = useParams();
+  const navigate = useNavigate();
+  const [product, setProduct] = useState(null);
+  const [comments, setComments] = useState([]);
+  const [newComment, setNewComment] = useState("");
+  const [isCommentValid, setIsCommentValid] = useState(false);
+
+  // 날짜 포맷 함수 (예: '2024-09-24 08:40:06' -> '2024-09-24 08:40')
+  function formatDate(dateString) {
+    const date = new Date(dateString);
+    return `${date.getFullYear()}-${(date.getMonth() + 1)
+      .toString()
+      .padStart(2, "0")}-${date.getDate().toString().padStart(2, "0")} ${date
+      .getHours()
+      .toString()
+      .padStart(2, "0")}:${date.getMinutes().toString().padStart(2, "0")}`;
+  }
+
+  // Fetch product details
+  useEffect(() => {
+    async function fetchProduct() {
+      try {
+        const data = await getProductDetails(productId);
+        setProduct(data); // 상품 데이터 설정
+      } catch (error) {
+        console.error("Error fetching product details:", error);
+      }
+    }
+
+    fetchProduct();
+  }, [productId]);
+
+  // Fetch comments
+  useEffect(() => {
+    async function fetchComments() {
+      try {
+        const data = await getProductComments(productId);
+        if (data && Array.isArray(data.list)) {
+          setComments(data.list); // list 안에 댓글 배열이 있을 경우
+        } else {
+          setComments([]); // list가 없으면 빈 배열 설정
+        }
+      } catch (error) {
+        console.error("Error fetching comments:", error);
+      }
+    }
+
+    fetchComments();
+  }, [productId]);
+
+  const handleCommentChange = (e) => {
+    const value = e.target.value;
+    setNewComment(value);
+    setIsCommentValid(value.trim().length > 0);
```

Comment:
처음부터 trim 을 해서 쓰면 더 좋을 거 같아요!

```suggestion
    const value = e.target.value.trim();
    setNewComment(value);
    setIsCommentValid(!!value);
```